File: templates/backend/python-fastapi/scaffold/app/routers/health.py
"""Health check router for container orchestration."""

from fastapi import APIRouter

router = APIRouter()


@router.get("/health")
async def health_check():
    """Return application health status.

    Used by container orchestrators (Kubernetes, ECS, Docker)
    to determine if the application is ready to receive traffic.
    """
    return {
        "status": "healthy",
        "checks": {
            "app": "ok",
        },
    }

File: templates/backend/node-express-api/scaffold/src/routes/health.ts
import { Router } from "express";

export const healthRouter = Router();

/**
 * Health check endpoint for container orchestration.
 * Returns application status and basic diagnostics.
 */
healthRouter.get("/", async (_req, res) => {
  res.json({
    status: "healthy",
    timestamp: new Date().toISOString(),
    checks: {
      app: "ok",
    },
  });
});
